indico/http/serialization.py

```diff
@@ -4,24 +4,43 @@
 from collections import defaultdict
 import cgi
 import json
+import io
+import gzip
 import logging
 import msgpack
 import traceback
 
-from indicoio.errors import IndicoDecodingError
+from indico.errors import IndicoDecodingError
 
 logger = logging.getLogger(__name__)
 
+def decompress(response):
+        response.raw.decode_content = True
+        value = io.BytesIO(response.raw.data).getvalue()
+        return gzip.decompress(value)
 
-def deserialize(response):
+
+def deserialize(response, gzip=False, force_json=False):
     content_type, params = cgi.parse_header(response.headers.get("Content-Type"))
+    
+    content = None
+    if gzip:
+        content = decompress(response)
+    else: 
+        content = response.content
+    
     charset = params.get("charset", "utf-8")
+
+    # For storage object for example where the content is json based on url ending
+    if force_json:
+        content_type = "application/json"
+
     try:
-        return _SERIALIZATINON_FNS[content_type](response.content, charset)
+        return _SERIALIZATINON_FNS[content_type](content, charset)
```

Comment:
Question about serialization msgpack. Since we don't install jetstream into the client, we won't have access to the full serialization of data that was generated and `packed` by a jetstream task. Are there any instances cases where the client is expecting to receive msgpack application data? There is a risk if we receive msgpack data but don't have the same deserialization ability of jetstream.